staff/views/batch.py

```diff
@@ -221,9 +225,44 @@ def post(self, request, batch_id):
 
         if basics_graduation_form.is_valid():
             enrolment_queryset = Enrolment.objects.filter(id__in=basics_graduation_form.cleaned_data.get('enrolment'))
-            enrolment_queryset.update(status=Enrolment.PASSED)
+            to_emails = []
 
-        return HttpResponse()
+            enrolment_queryset.update(status=Enrolment.PASSED)
+            for enrolment in enrolment_queryset:
+                student_user = enrolment.student_user
+                certificate = Certificate.objects.create(
+                    enrolment=enrolment,
+                    graduation_date=date.today()
+                )
+                referral_coupon = ReferralCoupon.objects.create(
+                    start_date=date.today(),
+                    referrer=enrolment.student_user
+                )
+                certificate_url = reverse(
+                    'basics_certificate',
+                    kwargs={'certificate_credential': certificate.credential}
+                )
+                print("!!!!!!!!!!!!!!!!!!!")
```

Comment:
Should we remove these prints before merging? Or are they being logged to some logging service that could be relevant for debugging in future?

Reply:
I'll remove them after testing with a batch of enrolments